torch/_inductor/codegen/cpp.py

```diff
@@ -3372,13 +3377,107 @@ def get_kernel_group(self):
         else:
             self.kernel_group = KernelGroup()
 
+    def fuse(self, node1, node2):
+        if node1.is_foreach() or node2.is_foreach():
+            return ForeachKernelSchedulerNode.fuse(node1, node2)
+        else:
+            if self._can_fuse_nodes_with_compatible_ranges(node1, node2):
+                assert isinstance(node1, (SchedulerNode, FusedSchedulerNode))
+                assert isinstance(node2, (SchedulerNode, FusedSchedulerNode))
+
+                _, (vars1, reduce1) = node1.group
+                _, (vars2, reduce2) = node2.group
+                assert reduce1 == () and reduce2 == (), (reduce1, reduce2)
+
+                def get_indexing_ranges_exprs(node):
+                    if isinstance(node, FusedSchedulerNode):
+                        assert len(node.snodes) > 0
+                        # use the last scheduler node from the list as it has the most
+                        # relevant indexing expressions
+                        return get_indexing_ranges_exprs(node.snodes[-1])
+                    else:
+                        assert isinstance(node, SchedulerNode)
+                        comp_buffer = node.node
+                        assert isinstance(comp_buffer, ir.ComputedBuffer)
+                        _, body, _ = comp_buffer.get_default_sizes_body()
+                        return body.var_ranges, [*body.indexing_exprs.values()]
+
+                node_to_recomp = node1 if len(vars1) < len(vars2) else node2
+                assert isinstance(node_to_recomp, SchedulerNode)
+
+                ref_node = node2 if len(vars1) < len(vars2) else node1
+
+                extra_indexing_constraints = get_indexing_ranges_exprs(ref_node)
+
+                node_to_recomp.recompute_size_and_body(
+                    extra_indexing_constraints=extra_indexing_constraints
+                )
+
+                if len(vars1) < len(vars2):
+                    node1 = node_to_recomp
+                else:
+                    node2 = node_to_recomp
```

Comment:
NIT: Isn't this redundant since `node_to_recomp` refers to the same object as `node1` or `node2`?